### PR TITLE
Swift Package Manager support

### PR DESCRIPTION
**Pull request by soffes**
Add `Package.swift`, fix some warnings, and move files to where SPM expects them.

### CODE REVIEW

**Comment by dmiluski** [comment]
Hi @soffes , so cool to see you here. Thanks for the contribution.

What would you think about including a swiftpm test as a part of the Travis CI?

eg. Referencing Travis step from [SDWebImage](https://github.com/SDWebImage/SDWebImage/blob/master/.travis.yml#L39)
```
    - echo Build the SwiftPM
    - swift build
    - rm -rf ~/.build
```

**Comment by soffes** [comment]
@dmiluski Hi! It's been awhile since I used Static :)

Trying to run `swift build` produces this error: `error: no such module 'UIKit'`. I could get it to build with `swift test -Xswiftc "-sdk" -Xswiftc "`xcrun --sdk iphonesimulator --show-sdk-path`" -Xswiftc "-target" -Xswiftc "x86_64-apple-ios13.0-simulator"`, but I couldn't figure out how to get SPM to run the tests.

Do you think it's still worth doing?

I'm using my fork via Xcode's SPM stuff in a project just fine for what it's worth 🤷

**Comment by dmiluski** [comment]
Perhaps not. 

I was trying to think if there was any way to prevent accidental breaks that may work in xcproj but would fail in the swiftpm at time of PR. 

eg.
Say if someone added a new file to the project root directory rather than Sources/ folder. But perhaps this is becoming more and more unlikely given Xcode Project often follows matching project folder/grouping?

@hyperspacemark any thoughts?

**Review by dmiluski — APPROVED** [review]

**Comment by dmiluski** [comment]
@factotvm @danhollywells Anybody have credentials to activate Travis CI?

File: Static.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -564,7 +580,8 @@
 		21826AC21B3F51A100AA9641 /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
-				INFOPLIST_FILE = Static/Tests/Info.plist;
+				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
```
Comment (keitaito) [inline Static.xcodeproj/project.pbxproj:583]: Would you mind elaborating on this change? Is it required for spm package support? The documentation says
> Always embed the Swift standard libraries in the target's products, even if the target does not contain any Swift code. For example, this should be enabled if the target is embedding other products which contain Swift, or if it is a test target which does not contain Swift but which is testing a product which does. This setting only applies to wrapped products, not to standalone binary products.

**Review by keitaito — APPROVED** [review]
Hi Sam, thanks for working on this! Left a few questions (sorry I'm totally a spm noob), but it looks good to me overall 🍏

**Comment by keitaito** [comment]
@dmiluski Hi Dane 👋 Will talk to DT and DHW and will figure out Travis CI.

File: Package.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// swift-tools-version:5.2
+
+import PackageDescription
+
+let package = Package(
+    name: "Static",
+    platforms: [
+        .iOS(.v9)
```
[the remark this answers is not in the document]
Comment (soffes) [reply Package.swift:8]: iOS 9.0

File: Static.xcodeproj/project.pbxproj (hunk 1 above)
Comment (soffes) [reply Static.xcodeproj/project.pbxproj:583]: Xcode did this automatically

File: Static.podspec [hunk 3]
```diff
@@ -8,7 +8,8 @@ Pod::Spec.new do |spec|
   spec.source = { git: 'https://github.com/venmo/Static.git', tag: "v#{spec.version}" }
   spec.author = { 'Venmo' => 'ios@venmo.com', 'Sam Soffes' => 'sam@soff.es' }
 
+  spec.swift_versions = ['5.0', '5.1', '5.2']
   spec.platform = :ios, '8.0'
```
Comment (kevnm67) [inline Static.podspec:12]: @soffes thoughts on matching the iOS versions? (e.g. `.v9` in SPM vs. `8.0` here).

**Comment by factotvm** [comment]
The failure on Travis CI is the following (abbreviated... hopefully the right parts):

```
Testing failed:
        Pattern cannot match values of type 'UITableView.Style'
        Testing cancelled because the build failed.
   
   Test session results and logs:
        /Users/travis/Library/Developer/Xcode/DerivedData/Static-hbdyaqfirynvrofangjcyubxdzok/Logs/Test/Test-Static-iOS-2020.07.19_01-00-15-+0000.xcresult
   
   ** TEST FAILED **
   
   
   The following build commands failed:
        CompileSwift normal x86_64 /Users/travis/build/venmo/Static/Sources/Static/DataSource.swift
        CompileSwiftSources normal x86_64 com.apple.xcode.tools.swift.compiler
   (2 failures)
```

I did pull down the code. I will admit I'm not well-versed with iOS projects with Swift Package Manager. For starters, `swift test` doesn't work out of the box. With a bit of finagling I got over UIKit not being found, but I gave up on getting the correct XCTest architecture involved.

Now, I _did_ run the tests in Xcode (there is no scheme for them inside Xcode, but creating one was trivial). Everything passed.

I see no reason not to get this code in. But, I have to get it passing CI first... call it a code of honor. So, the question is: why do the tests fail on Travis CI, but not locally? I noticed the Xcode version (`osx_image: xcode10.2`) is not up-to-date. Should that be updated to something more appropriate? Something that supports Swift Package Manager? The error doesn't seem particularly related ("Pattern cannot match values..."), but that's my only thought late on a Saturday night.

**Comment by factotvm** [comment]
Closed in favor of: https://github.com/venmo/Static/pull/156

**Pull request closed without merging**